apiserver/params/multiwatcher.go

```diff
@@ -14,20 +14,11 @@ import (
 
 	"github.com/juju/juju/core/constraints"
 	"github.com/juju/juju/core/instance"
+	"github.com/juju/juju/core/life"
 	"github.com/juju/juju/core/model"
 	"github.com/juju/juju/core/status"
 )
 
-// Life describes the lifecycle state of an entity ("alive", "dying"
```

Comment:
strictly speaking, each layer should define its own consts / data structures, and as info flows between layers, the data is transformed to the relevant representation for that layer.
+1 to the change to use core.Life in the api layer and its callers
- 1/2 to this bit since the purely correct approach is to map state.Life <-> params.Life in the apiserver layer and params.Life <-> core.Life in the api layer

Reply:
As discussed, we argree on having params types for structures, but not simple scalar types.